journal/src/main/java/io/camunda/zeebe/journal/file/SegmentedJournal.java

```diff
@@ -521,6 +524,30 @@ private boolean handleSegmentCorruption(
     return Arrays.asList(files);
   }
 
+  private void deleteDeferredFiles() {
+    try (final DirectoryStream<Path> segmentsToDelete =
+        Files.newDirectoryStream(
+            directory.toPath(),
+            path -> JournalSegmentFile.isDeletedSegmentFile(name, path.getFileName().toString()))) {
+      segmentsToDelete.forEach(
+          segmentFileToDelete -> {
+            try {
+              Files.deleteIfExists(segmentFileToDelete);
+            } catch (final IOException e) {
+              log.warn(
+                  "Could not delete file {} which is marked for deletion. This can result in unnecessary disk usage.",
+                  segmentFileToDelete,
+                  e);
```

Comment:
:upside_down_face: I think it would be more readable if we extract this lambda to its own method, `deleteDeferredFile`. That way the statement is then `segmentsToDelete.forEach(this::deleteDeferredFile)`.